webui/ocw/lib/EC2.py

```diff
@@ -1,23 +1,30 @@
+from .provider import Provider
+from .vault import EC2Credential
+from distutils.version import LooseVersion
+from dateutil.parser import parse
+from datetime import datetime
+from datetime import timedelta
+from datetime import timezone
 import boto3
+import re
 import time
-from .vault import EC2Credential
 import logging
 
+logger = logging.getLogger(__name__)
 
-class EC2:
+
+class EC2(Provider):
     __instances = dict()
     __key = None
     __secret = None
     __ec2_resource = dict()
     __ec2_client = dict()
     __credentials = None
-    __logging = None
 
     def __new__(cls, vault_namespace):
         if vault_namespace not in EC2.__instances:
             EC2.__instances[vault_namespace] = object.__new__(cls)
             EC2.__instances[vault_namespace].__credentials = EC2Credential(vault_namespace)
-            EC2.__instances[vault_namespace].__logging = logging.getLogger(__name__)
```

Comment:
maybe better keep it here and add vault_namespace to logger name ? so you can differentiate between log messages from different namespaces ?

Reply:
To advance for now. And I will keep it the same for all other providers.